companion.php

```diff
@@ -3,36 +3,53 @@
 Plugin Name: Companion Plugin
 Plugin URI: https://github.com/Automattic/companion
 Description: Helps keep the launched WordPress in order.
-Version: 1.31
+Version: 1.32
 Author: Osk
 */
 
 // Do a minimal set of stuff on a multisite installation on sites other than the main one
 if ( is_multisite() && ! is_main_site() ) {
 	add_action( 'pre_current_active_plugins', 'companion_hide_plugin' );
-	add_action( 'admin_notices', 'companion_admin_notices' );
+
+	if ( ! defined( 'IS_WP_STUDIO' ) ) {
+		add_action( 'admin_notices', 'companion_admin_notices' );
+	}
+
 	return true;
 }
 
 $companion_api_base_url = get_option( 'companion_api_base_url' );
 
-// Disable functionality if Atomic and not a JurassicNinja client site.
-if ( ! defined( 'IS_ATOMIC' ) || ( IS_ATOMIC && ( defined( 'IS_ATOMIC_JN' ) && IS_ATOMIC_JN ) ) ) {
+// Setup Hooks for WP.cloud
+if ( defined( 'IS_ATOMIC_JN' ) && IS_ATOMIC_JN ) {
 	add_action( 'wp_login', 'companion_wp_login', 1, 2 );
-	add_action( 'after_setup_theme', 'companion_after_setup_theme' );
-	add_action( 'admin_notices', 'companion_admin_notices' );
 	add_action( 'pre_current_active_plugins', 'companion_hide_plugin' );
+
+	if ( defined( 'IS_WP_STUDIO' ) && IS_WP_STUDIO ) {
+		// TODO : Add hooks for WP Studio sites.
```

Comment:
This is great, so we could add a custom `companion_admin_notices_studio` action to display our custom notice.